modules/orientdb/src/main/java/org/testcontainers/containers/OrientDBContainer.java

```diff
@@ -0,0 +1,181 @@
+
+    private static final int DEFAULT_BINARY_PORT = 2424;
+
+    private static final int DEFAULT_HTTP_PORT = 2480;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(OrientDBContainer.class);
+
+    private static final String DOCKER_IMAGE_NAME = DEFAULT_IMAGE_NAME + ":" + DEFAULT_TAG;
+
+    private String databaseName;
+
+    private String username;
+
+    private String password;
+
+    private OrientDB orientDB;
+
+    private ODatabaseSession session;
+
+    private Optional<String> scriptPath = Optional.empty();
+
+
+    public OrientDBContainer() {
+        this(DOCKER_IMAGE_NAME);
+    }
+
+    public OrientDBContainer(@NonNull String dockerImageName) {
+        super(dockerImageName);
+
+        username = DEFAULT_USERNAME;
+        password = DEFAULT_PASSWORD;
+        databaseName = DEFAULT_DATABASE_NAME;
+
+        WaitStrategy waitForHttp = new HttpWaitStrategy()
+            .forPort(DEFAULT_HTTP_PORT)
+            .forStatusCodeMatching(response -> response == HTTP_OK);
+
+        waitStrategy = new WaitAllStrategy()
+            .withStrategy(Wait.forListeningPort())
+            .withStrategy(waitForHttp)
+            .withStartupTimeout(Duration.ofMinutes(2));
+    }
+
+    @Override
+    protected void configure() {
+
+        addExposedPorts(DEFAULT_BINARY_PORT, DEFAULT_HTTP_PORT);
+        addEnv("ORIENTDB_ROOT_PASSWORD", "root");
+    }
+
+
+    public String getDatabaseName() {
+        return databaseName;
+    }
+
+    public String getUsername() {
+        return username;
+    }
+
+    public String getPassword() {
+        return password;
+    }
+
+    public String getTestQueryString() {
+        return "SELECT FROM V";
+    }
+
+    public OrientDBContainer withDatabaseName(final String databaseName) {
+        this.databaseName = databaseName;
+        return self();
+    }
+
+    public OrientDBContainer withUsername(final String username) {
+        this.username = username;
+        return self();
+    }
+
+    public OrientDBContainer withPassword(final String password) {
+        this.password = password;
+        return self();
+    }
+
+    public OrientDBContainer withScriptPath(String scriptPath) {
+
+        this.scriptPath = Optional.of(scriptPath);
+        return self();
+    }
+
+
+    @Override
+    protected void containerIsStarted(InspectContainerResponse containerInfo) {
+        orientDB = new OrientDB(getServerUrl(), "root", "root", OrientDBConfig.defaultConfig());
+    }
+
+    public OrientDB getOrientDB() {
+        return orientDB;
+    }
+
+    public String getServerUrl() {
+        return "remote:" + getContainerIpAddress() + ":" + getMappedPort(2424);
+    }
+
+    public String getDbUrl() {
+        return getServerUrl() + "/" + databaseName;
+    }
+
+    public ODatabaseSession getSession() {
+
+        orientDB.createIfNotExists(databaseName, ODatabaseType.PLOCAL);
+
+        if (session == null) {
```

Comment:
If `getSession()` is called concurrently, couldn't we end up loading the script multiple times?

I think it might be safer to just have this method be `synchronized`.

Reply:
done